### DISCUSSION TITLE
Arduino Uno R3 Client driver Interface

### DISCUSSION THREAD

**Discussion by JaviersitoSF** [arendst/Tasmota#22024] [Support request] [answered]
I am having an issue when upload the .hex file to upgrade the Arduino Uno R3, it says "Upload Failed Invalid file signature" when I try to install the "AnalogJSON.ino.hex" I am following the steps of the webpage https://tasmota.github.io/docs/TasmotaClient/#configuring-tasmota-to-use-the-tasmotaclient-functionality

I have double checked the connections on my Wemos D1 mini F12 with Esp8266 to the Arduino uno R3

Wemos D1 F12, functions perfect with the Tasmota, console recognizes commands, wifi connected, and upgraded to 14.2.0 (release-tasmota-4M), all nice with Wemos D1 F12

All pins connected using  a logic board to modulate from 5 to 3.3 volts the signals that come from  Arduino to the Wemos D1 F12 and viceversa.

the generated compiled file is created witth Arduino IDE as per instructions of the after mentioned webpage.

Anyone can help me, please, I am really new to Tasmota.  Thank you in advance.

Javier Sánchez

**Comment by Jason2866** [comment, marked as answer]
The feature Tasmota Client is not included in a precompiled firmware. You have to compile Tasmota yourself to include this feature.

**Reply by JaviersitoSF** [reply]
Thank you!

**Comment by JaviersitoSF** [comment]
Thank you!
Can you please direct me to the instructions within Trasmota, on how to
communicate bidirectionally with Arduino, the AT commands do not work and
happily I found Tasmota that has allowed me to use the Wemos D1 mini F12,
but I need to send data with a web server for my final class work using an
Arduino Uno.

I'm sorry to bother you, just point me, I can not find it.

Javier

El lun, 26 ago 2024 a las 11:38, Jason2866 ***@***.***>)
escribió:

> The feature Tasmota Client is not included in a precompiled firmware. You
> have to compile Tasmota yourself to include this feature.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/arendst/Tasmota/discussions/22024#discussioncomment-10454557>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BA6SNSLI2PDKXE7WI2UY3ZLZTNRYZAVCNFSM6AAAAABNEOC6CKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTANBVGQ2TKNY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Javier Sánchez F.

Tel. (502)2416-1616

**Reply by sfromis** [reply]
You already linked the page with the instructions, which includes a link to Github project with examples, and a library with a number of send functions.
https://tasmota.github.io/docs/TasmotaClient/
